src/arithmetic/algebraic_expression.c

```diff
@@ -261,6 +260,23 @@ void AlgebraicExpression_PrependTerm(AlgebraicExpression *ae, GrB_Matrix m, bool
     ae->operands[0].operand = m;
 }
 
+void AlgebraicExpression_Transpose(AlgebraicExpression *ae) {
+
+    // Switch expression src and dest nodes.
+    Node **n = ae->src_node;
+    ae->src_node = ae->dest_node;
+    ae->dest_node = n;
+
+    if(ae->operand_count == 1) {
+        // Expression transpose is determined by its only term.
+        ae->transpose = false;
+        ae->operands[0].transpose = !ae->operands[0].transpose;
+    } else {
+        // Flip expression transpose flag.
+        ae->transpose = !ae->transpose;
+    }
+}
+
 void AlgebraicExpression_Free(AlgebraicExpression* ae) {
```

Comment:
This function probably needs to free duplicated matrices while leaving originals untouched. I _think_ that right now we can safely free any operand with its transpose flag set?

Reply:
The only case were we allocate a new matrix, is when an expression has a single term in it, and the term is transposed.
When that happens the term is considered the expression result, and that get freed in AlgebraicExpressionResult_Free

Other terms which "get" transposed are only transposed in the context on a multiplication,
they're aren't really transposed and we're not allocating new matrices for them,

So I think we're good here.